standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/parser/Filter.g

```diff
@@ -549,7 +549,6 @@ DateLiteral
 TimestampLiteral
     :
     KW_TIMESTAMP '\'' TimestampString '\'' { extractTimestamp(getText()) != null }?
-    | TimestampString { extractTimestamp(getText()) != null }?
```

Comment:
It is possible with a slight change in you proposed rule.
Since the unquoted parts conflicts I had to move the unquoted ts branch to always execute predicate.

Reply:
I now understand better the problem with the conflict; I forgot that rule order does not matter.